edx_analytics_transformers/utils/http_client.py

```diff
@@ -0,0 +1,56 @@
+"""
+A generic HTTP Client.
+"""
+from logging import getLogger
+
+import requests
+
+
+logger = getLogger(__name__)
+
+
+class HttpClient:
+    """
+    A generic HTTP Client.
+    """
+    def __init__(self, url='', auth_scheme='', auth_key='', headers=None):
+        """
+        Initialize the client with provided configurations.
+
+        url (str)        :     URL for the event consumer.
+        auth_scheme (str) :     Scheme used for authentication.
+        auth_key (str)    :     API key used in the authorization header.
+        headers (str)     :     Any additional headers to be sent with event payload.
+        """
+        self.URL = url
+        self.AUTH_SCHEME = auth_scheme
+        self.AUTH_KEY = auth_key
+        self.HEADERS = headers or {}
+
+    def get_auth_header(self):
+        """
+        Generate auth headers depending upon the client configurations.
+
+        Returns:
+            dict
+        """
+        if self.AUTH_SCHEME:
+            return {
+                'Authorization': '{} {}'.format(self.AUTH_SCHEME, self.AUTH_KEY)
+            }
+        return {}
+
+    def send(self, json):
+        """
+        Send the event to configured remote.
+
+        Arguments:
+            json (dict) :   event payload to send to host.
+
+        Returns:
+            requests.Response object
+        """
+        headers = self.HEADERS.copy()
+        headers.update(self.get_auth_header())
+        logger.info('Sending event json to %s', self.URL)
+        return requests.post(self.URL, json=json, headers=headers)
```

Comment:
This should have a configurable timeout so that it does not hang forever.

Reply:
I've added the support for providing parameters through configurations.